@Deedone pls, fix compliance checks 
```

 C99_COMMENTS: do not use C99 // comments
File:subsys/fs/shell.c
Line:37
Error: Process completed with exit code 1.
```

--- a/drivers/sdhc/rcar_mmc.c
+++ b/drivers/sdhc/rcar_mmc.c
@@ -2248,6 +2248,14 @@ static int rcar_mmc_init(const struct device *dev)
 		return -EFAULT;
 	}
 
+#ifdef CONFIG_RCAR_MMC_DMA_IRQ_DRIVEN_SUPPORT
+	ret = k_sem_init(&data->irq_xref_fin, 0, 1);
+	if (ret) {
+		LOG_ERR("%s: can't init semaphore", dev->name);
+		return ret;
+	}
+#endif
+
 	if (unlikely(dev->data == NULL)) {
 		LOG_ERR("%s: there isn't data instance inside device structure", dev->name);
 		return -EFAULT;
@@ -2289,11 +2297,6 @@ static int rcar_mmc_init(const struct device *dev)
 	}
 
 #ifdef CONFIG_RCAR_MMC_DMA_IRQ_DRIVEN_SUPPORT
-	ret = k_sem_init(&data->irq_xref_fin, 0, 1);
-	if (ret) {
-		LOG_ERR("%s: can't init semaphore", dev->name);
-		goto exit_disable_clk;
-	}
 	cfg->irq_config_func(dev);
 #endif /* CONFIG_RCAR_MMC_DMA_IRQ_DRIVEN_SUPPORT */
 

--- a/subsys/fs/littlefs_fs.c
+++ b/subsys/fs/littlefs_fs.c
@@ -856,7 +856,7 @@ static int littlefs_init_cfg(struct fs_littlefs *fs, int flags)
 	return 0;
 }
 
-static int littlefs_init_fs(struct fs_littlefs *fs, int dev_id, int flags)
+static int littlefs_init_fs(struct fs_littlefs *fs, uintptr_t dev_id, int flags)
 {
 	int ret = 0;
 

--- a/subsys/fs/shell.c
+++ b/subsys/fs/shell.c
@@ -33,6 +33,17 @@ static struct fs_mount_t fatfs_mnt = {
 /* LITTLEFS */
 #ifdef CONFIG_FILE_SYSTEM_LITTLEFS
 #include <zephyr/fs/littlefs.h>
+
+/* TODO: Implement dynamic storage dev selection */
+#ifdef CONFIG_FS_LITTLEFS_BLK_DEV
+FS_LITTLEFS_DECLARE_DEFAULT_CONFIG(lfs_data);
+static struct fs_mount_t littlefs_mnt = {
+	.type = FS_LITTLEFS,
+	.fs_data = &lfs_data,
+	.flags = FS_MOUNT_FLAG_USE_DISK_ACCESS,
+	.storage_dev = CONFIG_MMC_VOLUME_NAME,
+};
+#else
 #include <zephyr/storage/flash_map.h>
 
 FS_LITTLEFS_DECLARE_DEFAULT_CONFIG(lfs_data);
@@ -41,6 +52,7 @@ static struct fs_mount_t littlefs_mnt = {
 	.fs_data = &lfs_data,
 	.storage_dev = (void *)STORAGE_PARTITION_ID,
 };
+#endif
 #endif
 
 #define BUF_CNT 64
@@ -523,6 +535,15 @@ static int cmd_mount_fat(const struct shell *shell, size_t argc, char **argv)
 
 static int cmd_mount_littlefs(const struct shell *shell, size_t argc, char **argv)
 {
+#ifdef CONFIG_FS_LITTLEFS_BLK_DEV
+	static const char *disk_mount_pt = "/"CONFIG_MMC_VOLUME_NAME":";
+
+	char *mntpt = mntpt_prepare(argv[1]);
+
+	if (!mntpt) {
+		mntpt = disk_mount_pt;
+	}
+#else
 	if (littlefs_mnt.mnt_point != NULL) {
 		return -EBUSY;
 	}
@@ -533,6 +554,7 @@ static int cmd_mount_littlefs(const struct shell *shell, size_t argc, char **arg
 		shell_error(shell, "Failed to allocate mount point");
 		return -ENOEXEC; /* ?!? */
 	}
+#endif
 
 	littlefs_mnt.mnt_point = mntpt;
 

